File: tutorials/testing/java/game_snippet.java
import java.util.List;
import java.util.Random;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.Game;

// this program fragment illustrates how to access the IGN Games data
public class game_snippet {
	public static void main(String[] args) throws Exception {

		//create the Bridges object
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

		/* Get a List of Game objects from Bridges */
		DataSource ds = bridges.getDataSource();
		List<Game> mylist = ds.getGameData();

		// Inspect a random Game object
		Game game1 = mylist.get((new Random()).nextInt(mylist.size()));
		System.out.println(game1.getTitle());
		System.out.println(game1.getPlatformType());
		System.out.println(game1.getRating());
		System.out.println(game1.getGenre());
	}
}

File: tutorials/testing/c++/game_snippet.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/Game.h"


using namespace bridges;


// This program fragment illustrates how to access the IGN Game data
int main(int argc, char **argv) {

	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");

	// set title
	bridges.setTitle("How to access the IGN Game Data");

	// read the IGN game data
	DataSource ds (&bridges);
	vector<Game> game_list = ds.getGameData();

	// Print a single record of the data for illustration
	cout << "Game 0:" << endl;
	cout << "\tTitle: " << game_list[0].getTitle() << endl
		<< "\tPlatform Type: " << game_list[0].getPlatformType() << endl
		<< "\tRating: " << game_list[0].getRating() << endl <<
		"\tGenres: ";
	for (auto& s : game_list[0].getGameGenre())
		cout << s << "," << endl;

	return 0;
}
